python/external_config/external_command.py

```diff
@@ -59,13 +61,44 @@ def serialize_command(cls, entity_type, command_name, properties):
             "group_default": properties.get("group_default") or False,
 
             # special for shotgun
-            "deny_permissions": properties.get("deny_permissions"),
-            "deny_platforms": properties.get("deny_platforms"),
-            "supports_multiple_selection": properties.get("supports_multiple_selection"),
+            "sg_deny_permissions": properties.get("deny_permissions"),
+            "sg_supports_multiple_selection": properties.get("supports_multiple_selection"),
         }
 
         return data
 
+    @classmethod
+    def enabled_on_current_os(cls, properties):
+        """
+        Checks toolkit command properties to determine
+        if a command is enabled on the current os or Not.
+
+        :param dict properties: Properties dictionary
+            as returned by the ``Engine.commands`` dictionary property.
+        :return: True if enabled, False if not.
+        """
+        if "deny_platforms" in properties:
+            # setting can be Linux, Windows or Mac
+            curr_os = {"linux2": "Linux", "darwin": "Mac", "win32": "Windows"}[sys.platform]
+            if curr_os in properties["deny_platforms"]:
+                # not enabled on this platform
+                return False
+
+        return True
+
+    @classmethod
+    def is_compatible(cls, data):
+        """
+        Determines if the given data is compatible.
+
+        :param dict data: Serialized data
+        :returns: True if the given data can be loaded, False if not.
+        """
+        try:
```

Comment:
Is it some kind of "ask for forgiveness" pattern? If so, what is the exception we expect here? We use a `get` so we can't have a `KeyError`, so it seems to only error could come from `data` not being a dictionary?

Reply:
Changing to `KeyError`!